protobuf-lite/BUILD.bazel

```diff
@@ -3,12 +3,11 @@ java_library(
     srcs = glob([
         "src/main/java/**/*.java",
     ]),
-    # TOOD(zdapeng): fix visibility and deps (https://github.com/google/protobuf/issues/2762)
-    visibility = ["//protobuf:__pkg__"],
+    visibility = ["//visibility:public"],
     deps = [
         "//core",
         "@com_google_code_findbugs_jsr305//jar",
         "@com_google_guava_guava//jar",
-        "@com_google_protobuf//:protobuf_java",
+        "@com_google_protobuf_javalite//:protobuf_java_lite",
```

Comment:
So the problem is that this breaks non-lite; you can't mix the two runtimes. We need to have a `select()` to auto-choose the correct dependency (lite on android, otherwise full protobuf) and I've not had a chance to get back and try some options.

Reply:
I didn't think it would use both runtimes, because it's the `protobuf-lite` folder. 
But there is a possible conflict here 
https://github.com/grpc/grpc-java/blob/master/java_grpc_library.bzl#L112 
https://github.com/grpc/grpc-java/blob/master/protobuf/BUILD.bazel#L9
But I have not much experience with this repository, so not sure, so I'll leave this to you.
We decided to move away from lite to have any support.